### ISSUE TITLE
JWT decode callback token exception

### ISSUE THREAD

**Issue by handleror** [zquestz/omniauth-google-oauth2#218]
got this issue after omniauth callback results:
``
JWT::InvalidIatError (Invalid iat):
  jwt (1.5.0) lib/jwt.rb:170:in `decode'
  omniauth-google-oauth2 (0.2.8) lib/omniauth/strategies/google_oauth2.rb:61:in `block in <class:GoogleOauth2>'
``

**Comment by Ashviniv** [comment]
I got the same error.I skipped JWT decoding by adding scope ```skip_jwt: true``` in the omniauth configuration file i.e omniauth.rb.
```
Rails.application.config.middleware.use OmniAuth::Builder do
  provider :google_oauth2, GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET, {scope: 'email', skip_jwt: true}
end
```

**Comment by handleror** [comment]
Thank @Ashviniv, I just found in these issues
https://github.com/zquestz/omniauth-google-oauth2/issues/195
https://github.com/zquestz/omniauth-google-oauth2/pull/196
Guest it should be closed

**Comment by Shwetakale** [comment]
@Ashviniv @handleror Do you know root cause of this issue? I am facing same issue and behavior is total random. Sometimes I am successfully able to login and sometimes get this error. I observed such errors in my production log as well and those are very random or do anyone of you know reproduction steps?

**Comment by Ashviniv** [comment]
@Shwetakale As per my knowledge and the closed issue [#197](https://github.com/zquestz/omniauth-google-oauth2/issues/197) omniauth-google-oauth2 uses  [JWT gem,](https://github.com/jwt/ruby-jwt) which contains the ruby implementation of oauth algorithm. which internally uses `JWT.decode `to ensure that an attacker cannot bypass the algorithm verification step.
It uses `iat timestamp` and if there is timestamp mismatch this error comes. With the option `skip_jwt: true`. we avoid calling `JWT.decode`.

I hope the  Issue #197 can help you to reproduce the issue.
